cognite/client/_api/simulators/models.py

```diff
@@ -0,0 +1,430 @@
+                >>> from cognite.client import CogniteClient
+                >>> client = CogniteClient()
+                >>> res = client.simulators.models.revisions.list()
+
+            Specify filter and sort order:
+                >>> from cognite.client.data_classes.simulators.filters import SimulatorModelRevisionsFilter
+                >>> res = client.simulators.models.revisions.list(
+                    ...     filter=SimulatorModelRevisionsFilter(model_external_ids=["model_external_id"])
+                    ...     sort=CreatedTimeSort(order="asc")
+                    ... )
+        """
+        self._warning.warn()
+        return self._list(
+            method="POST",
+            limit=limit,
+            resource_cls=SimulatorModelRevision,
+            list_cls=SimulatorModelRevisionList,
+            sort=[CreatedTimeSort.load(sort).dump()] if sort else None,
+            filter=filter.dump(camel_case=True) if isinstance(filter, CogniteFilter) else filter,
+        )
+
+    @overload
+    def retrieve(self, id: int | None = None, external_id: str | None = None) -> SimulatorModelRevision | None: ...
+
+    @overload
+    def retrieve(
+        self,
+        id: int | Sequence[int] | None = None,
+        external_id: str | SequenceNotStr[str] | None = None,
+    ) -> SimulatorModelRevision | SimulatorModelRevisionList | None: ...
+
+    def retrieve(
+        self,
+        id: int | Sequence[int] | None = None,
+        external_id: str | SequenceNotStr[str] | None = None,
+    ) -> SimulatorModelRevision | SimulatorModelRevisionList | None:
+        """`Retrieve simulator model revision(s) <https://developer.cognite.com/api#tag/Simulator-Models/operation/retrieve_simulator_model_revisions_simulators_models_revisions_byids_post>`_
+        Retrieve one or more simulator model revisions by ID(s) or external ID(s)
+        Args:
+            id (int | Sequence[int] | None): The ids of the simulator model revisions.
+            external_id (str | SequenceNotStr[str] | None): The external ids of the simulator model revisions.
+        Returns:
+            SimulatorModelRevision | SimulatorModelRevisionList | None: Requested simulator model revision(s)
+        Examples:
+            Get simulator model revision by id:
+                >>> from cognite.client import CogniteClient
+                >>> client = CogniteClient()
+                >>> res = client.simulators.models.revisions.retrieve(id=1)
+
+            Get simulator model revision by external id:
+                >>> res = client.simulators.models.revisions.retrieve(external_id="revision_external_id")
+
+            Get multiple simulator model revisions by ids:
+                >>> res = client.simulators.models.revisions.retrieve(ids=[1,2])
+
+            Get multiple simulator model revisions by external ids:
+                >>> res = client.simulators.models.revisions.retrieve(external_ids=["revision1", "revision2"])
```

Comment:
See previous comment, these should raise in doctests
```suggestion
                >>> res = client.simulators.models.revisions.retrieve(external_id=["revision1", "revision2"])
```

Reply:
Please check if there are more occurrences of the above

Reply:
Fixed